jwst/background/background_sub.py

```diff
@@ -295,7 +295,9 @@ def subtract_wfss_bkg(input_model, bkg_filename, wl_range_name, mmag_extract=Non
     # i.e. in regions we can use as background.
     if got_catalog:
         bkg_mask = mask_from_source_cat(input_model, wl_range_name, mmag_extract)
-        if bkg_mask.sum() < 100:
+        # Ensure mask has 100 pixels and that those pixels correspond to valid
+        # pixels using model DQ array
+        if bkg_mask.sum() < 100 or sum(input_model.dq[bkg_mask] % 2 ^ 1) < 100:
```

Comment:
I'd rather see `dqflags.pixel['DO_NOT_USE']` used in this expression, rather than hardwiring a numerical value.

Reply:
✅